### PR TITLE
Add local file deletion on `torrent.destroy`

### PR DESCRIPTION
**Pull request by Rowern**
When we call `torrent.destroy` and we are not running in a browser, we use
`rimraf` to delete the folder containing the torrent files.

This is a work in progress because I need feedback on a good way to implement it.
For instance, `client.destroy` calls `torrent.destroy` on all its torrents, this might be problematic as destroying the client looks like the legitimate way of stopping every torrent transactions but keeping your progress.

### CODE REVIEW

**Comment by feross** [comment]
Thanks for the PR, @Rowern!

I think the right way to handle this is to call the `destroy()` method on the store, which already uses `rimraf` internally. See here: https://github.com/feross/fs-chunk-store/blob/8d7a57b56572625346024fca8cf3e2839ba2d49d/index.js#L222-L232

I haven't looked, but I suspect that we're only calling `close()` to close the store, but not destroy it.

You're right that we also want to keep the option to destroy the torrent without destroying the files on disk, so that probably should be an option to `torrent.destroy()`, like `torrent.destroy({ remove: true })` which defaults to `false`.

**Comment by Rowern** [comment]
You are right, I will continue to work on this PR tomorrow.

File: test/torrent-destroy.js [hunk 1]
```diff
@@ -22,3 +24,34 @@ test('torrent.destroy: destroy and remove torrent', function (t) {
...
+  }, function (torrent) {
+    t.equal(client.torrents.length, 1)
+    t.equal(torrent.infoHash, fixtures.leaves.parsedTorrent.infoHash)
+    t.equal(torrent.magnetURI, fixtures.leaves.magnetURI)
+
+    var completeFileName = path.join(torrent.path, torrent.files[0].name)
+
+    client.remove(torrent, {'remove': true}, function (err) {
+      t.error(err, 'torrent removed')
+      fs.stat(completeFileName, function (err) {
```
Comment (Rowern) [inline test/torrent-destroy.js]: I know this might not work well with browser API, I might add a check on the type of `store` used.

**Comment by Rowern** [comment]
Sorry for closin/re-opening but it was the only way to re-trigger a travis build.

**Pull request closed without merging**